File: app/src/main/java/com/digzdigital/divinitytoday/data/AppDataManager.java
package com.digzdigital.divinitytoday.data;

import com.digzdigital.divinitytoday.DivinityTodayApp;
import com.digzdigital.divinitytoday.data.db.DbHelper;
import com.digzdigital.divinitytoday.data.wp.WpHelper;

import javax.inject.Inject;


public class AppDataManager implements DataManager {
    @Inject
    public WpHelper wpHelper;
    @Inject
    public DbHelper dbHelper;

    public AppDataManager() {
        DivinityTodayApp.getInstance().getAppComponent().inject(this);

    }

    @Override
    public void setWpListener(WpListener listener) {
        wpHelper.setWpListener(listener);
    }

    @Override
    public void queryForOnlinePosts(int offset) {
        wpHelper.queryForOnlinePosts(offset);
    }

    @Override
    public void setDbListener(DbListener listener) {
        dbHelper.setDbListener(listener);
    }

    @Override
    public void savePost(Devotional devotional) {
        dbHelper.savePost(devotional);
    }

    @Override
    public void queryForPosts() {
        dbHelper.queryForPosts();
    }

    @Override
    public void deletePost(Devotional devotional) {
        dbHelper.deletePost(devotional);
    }
}

File: app/src/main/java/com/digzdigital/divinitytoday/data/AppDataManager.kt
package com.digzdigital.divinitytoday.data

import com.digzdigital.divinitytoday.data.db.DbHelper
import com.digzdigital.divinitytoday.data.model.Devotional
import com.digzdigital.divinitytoday.data.wp.WpHelper
import rx.Observable

class AppDataManager(val dbHelper: DbHelper,val wpHelper: WpHelper): DataManager {
    override fun getDevotionals(offset: String, per_page: String): Observable<List<Devotional>> {
        return wpHelper.getDevotionals(offset, per_page)
    }

    override fun savePost(devotional: Devotional): Boolean {
        return dbHelper.savePost(devotional)
    }

    override fun queryForPosts(): Observable<List<Devotional>> {
        return dbHelper.queryForPosts()
    }

    override fun deletepost(devotional: Devotional) {
        dbHelper.deletepost(devotional)
    }
}
